quic/s2n-quic/Cargo.toml

```
@@ -91,3 +91,8 @@ s2n-quic-transport = { path = "../s2n-quic-transport", features = ["unstable_res
 tokio = { version = "1", features = ["full"] }
 tracing = { version = "0.1" }
 tracing-subscriber = { version = "0.3", features = ["env-filter"] }
+
+# Quiche's dependency BoringSSL can't be built on i686-unknow-linux-gnu
+# Hence, we need to skip importing quiche on that platform
+[target.'cfg(not(target_arch = "x86"))'.dev-dependencies]
+quiche = "0.24"
```

```suggestion
quiche = { version = "0.24", optional = true }
```
Does this work instead? This is a little cleaner and keeps the crate from pulling in quiche if the module utilizing quiche is compiled out.

---

I don't think so. We only use `quiche` in one of our tests, so it has to be a dev-dependency. A dev-dependency can not be optional:
```
EC2:~/workspace/s2n-quic$ cargo test
error: failed to load manifest for workspace member `/home/ubuntu/workspace/s2n-quic/quic/s2n-quic`
referenced via `quic/s2n-*` by workspace at `/home/ubuntu/workspace/s2n-quic/Cargo.toml`

Caused by:
  failed to parse manifest at `/home/ubuntu/workspace/s2n-quic/quic/s2n-quic/Cargo.toml`

Caused by:
  dev-dependencies are not allowed to be optional: `quiche`
```